### ISSUE TITLE
No CSR-enabled webparts found on the page

### ISSUE THREAD

**Issue by kamnevoleg** [andrei-markeev/cisar#22]
Hello! 
I've trying to simply customize list view via JSLink. Its not working! I activated/deactivate the MDS feature and always the same thing, not working!
To call the jslink script on the webpart, I've linked to: ~site/siteassets/subject_template.js or ~sitecollection/siteassets/subject_template.js
I installed the Cisar Chrome Extension (trying install from google and zip-archive .) - but it's not working! 
F12 on view, forms -> Cisar -> "No CSR-enabled webparts found on the page"

I've seen discussion here "https://github.com/andrei-markeev/cisar/issues/21" but but nothing helped.
What requirements on SP2013 on-premise must be met for Cisar? (production server:MDF is on, test server: MDF is off)
I checked on the production and on the test server - the same result. 

Unfortunately the problem is possible in the SP2013, but visually it works, except JSLink or CSR.

Thank you in advance.

**Comment by andrei-markeev** [comment]
Hi,
First of all, are you sure you have CSR-enabled webparts on the page?
For example, new lists experience cannot be customized by CSR. Also Calendar lists are not supported.
In case of new lists experience, as far as I know where is no official way currently to customize them at all, so the only way is to switch back to classic SharePoint experience...

Best,
Andrei

**Comment by kamnevoleg** [comment]
Of course, I checked on new task list, custom list, DisplayForm, EditForm, listview, homepage. I have tested 2 PCs, 2 SPS2013 and different Chrome profiles. Everywhere Cisar not recognizing the webparts and displays  "No CSR-enabled webparts found on the page".

**Comment by Danny-Engelman** [comment]
You say Client Side Rendering is not working.

Cisar makes CSR development easier, it is not required for CSR development.

Without Cisar you can set JSLinks on WebParts, you can create JS files.

So before we go an a Cisar wild-goose chase (it works fine in my Online environment)
First do Client Side Rendering the manual way, and verify you don't get the concept wrong. 

Create a simple JS file with only:

``
    console.log('JSLink executed my CSR file');
``
And point a JSLink on a WebPart to it

If that works

you then open Cisar, and it should recognize the JSLink

Now, it could well be Andrei created a bug when fixing the latest WP1 issue,
it is a side project for him so don't expect him to test every old SP version

**Comment by andrei-markeev** [comment]
Hi,

I'm actually pretty confident it should work. I improved the way Cisar recognizes webparts several weeks ago, and been using it quite intensively over last weeks and haven't noticed any more issues with that.

I checked just now both on-prem installation of SP2013 and Office365 and Cisar is able to find CSR-enabled webparts without issues. I checked both list views and list forms.

Again - it will not work with new library/list experience or with Calendar lists, but otherwise should work fine.

Oleg, when you experience this issue, could you press Ctrl+Shift+I from Cisar - this will open second DevTools that can be used to debug the main DevTools window :) And check if there are some errors in the console in there?

Thanks,
Andrei

**Comment by kamnevoleg** [comment]
On test custom list JSLink works ok, connected script on the web of the list by the way ~site/SiteAssets/TestColor.js

![_1](https://cloud.githubusercontent.com/assets/19527462/20885300/c9d22750-baf8-11e6-8a2a-899c6225e10a.JPG)

After that, I open Cesar, then left panel constantly hangs waiting:

![_2](https://cloud.githubusercontent.com/assets/19527462/20885318/de37cbc8-baf8-11e6-855d-e620c56f2123.JPG)

When I press Ctrl+Shift+I from Cisar, nothing opens.

**Comment by andrei-markeev** [comment]
Ctrl+Shift+I is a standard Chrome functionality - it's not related to Cisar.
According to the [docs](https://developers.google.com/web/tools/chrome-devtools/shortcuts), it seems that it only works when DevTools are undocked.

Please undock and try again.

What you should have after Ctrl+Shift+I is something like this:

![image](https://cloud.githubusercontent.com/assets/5181724/20885584/5c0f5d4e-bafa-11e6-9601-51f4e9486035.png)

**Comment by kamnevoleg** [comment]
![image](https://cloud.githubusercontent.com/assets/19527462/20886330/3b387cdc-bafe-11e6-996f-01d7741039ae.png)

**Comment by andrei-markeev** [comment]
This is really mysterious.

Some checks that you still can try:
 - Do you have some sort of AdBlocker installed? Can you switch it off temporarily and try using Cisar again? (Cisar doesn't load any ads or any external URLs at all, but maybe there was some false-positive still).
 - Maybe you have some other extensions installed on both computers where you tried Cisar, that might interfere?
 - In [chrome://extensions](chrome://extensions), Cisar version should be 2.0.3

**Comment by kamnevoleg** [comment]
After disabling all extensions and reload Chrome:

![image](https://cloud.githubusercontent.com/assets/19527462/20888827/18975ce2-bb09-11e6-9293-72cae621513c.png)

Sorry about that

**Comment by andrei-markeev** [comment]
So does it work now or still same problem?

**Comment by kamnevoleg** [comment]
still same problem

**Comment by kamnevoleg** [comment]
To clear the cache, delete the extensions did not solve the problem, only after reinstalling chrome - problem is solved.

**Comment by andrei-markeev** [comment]
Really strange issue. But glad it is finally solved!
Please don't hesitate to report any other bugs. Thanks! :)
